### PR TITLE
Add file deletion to Code surface and improve shelf collapse UX

### PR DESCRIPTION
**Pull request by gtanczyk**
## Summary
This PR adds the ability to delete files from the Code surface with a two-click confirmation pattern, and improves the studio shelf collapse behavior to support manual collapsing on shorter shelves.

### Code Surface File Deletion
- **Backend API**: Added `POST /api/me/studio/games/:slug/sources/stage/delete` endpoint that stages a file deletion marker, preventing the file from appearing in the next staged rebuild
- **Frontend UI**: 
  - Restructured file picker options from single buttons to containers with separate "open" and "delete" buttons
  - Implemented two-click confirmation pattern (first click "arms" the delete button, second click confirms deletion)
  - Delete button only appears for editable code surfaces
  - Added visual feedback with armed state styling and error color indicators
- **API Client**: Added `deleteCodeSurfaceFile()` function to call the new deletion endpoint
- **Tests**: Added comprehensive test coverage for the deletion flow and edge cases (agent round conflicts, closed rounds, permission checks)

### Studio Shelf Collapse Improvements
- **Manual Collapse**: Added `shelfCollapsedByUser` state to allow creators to manually collapse short shelves (below `STUDIO_SHELF_TOOLS_AT` threshold) via the edge toggle button
- **Unified Behavior**: Both auto-collapsed (long shelves) and manually-collapsed (short shelves) shelves now use the same `is-compact-shelf` styling
- **Smart Toggle**: 
  - Edge toggle on short shelves collapses to rail; on long shelves it opens/closes the overlay
  - Summary button on manually-collapsed shelves restores full shelf directly instead of opening overlay
- **CSS Updates**: Added rule to show edge toggle on desktop when game is open and shelf isn't already auto-compacted
- **Tests**: Added test verifying manual collapse behavior on short shelves

### UI/UX Refinements
- Refactored `closeFilePicker()` helper to consolidate file picker closing logic
- Updated file option styling to accommodate the new delete button layout
- Added internationalization strings for delete actions (English and Polish)

## Implementation Details
- File deletion respects the same round lifecycle as other staging operations (refuses during live agent rounds, opens new manual rounds if needed)
- Delete operations clear any pending save timers for the file before deletion
- After successful deletion, the UI automatically selects the next available file or clears selection if none remain
- The two-click pattern follows the same UX pattern as the admin job cancel button for consistency

https://claude.ai/code/session_018t1draE6PFUKWqkY6Ams71

### CODE REVIEW

**Comment by gtanczyk** [comment]
@codex could you take a look at this PR and share feedback?

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/src/CodeSurface.tsx [hunk 1]
```diff
@@ -889,6 +899,43 @@ export function CodeSurface({
...
+    try {
+      await deleteCodeSurfaceFile(slug, path);
+      setDrafts((current) => {
+        if (!(path in current)) return current;
+        const next = { ...current };
+        delete next[path];
+        return next;
+      });
+      const result = await fetchCodeSurfaceSources(slug);
+      setSources(result);
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/web/src/CodeSurface.tsx:939]: Good catch — fixed in 17c3415: `changedPaths` now folds `sources.deleted` into the working-copy set, so a delete-only change keeps Discard/Publish available.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/src/CodeSurface.tsx [hunk 2]
```diff
@@ -889,6 +899,43 @@ export function CodeSurface({
...
+    }
+    if (deletingPath) return;
+    setDeleteArmedPath(null);
+    setDeletingPath(path);
+    setDeliverMessage(null);
+    const timer = saveTimersRef.current.get(path);
+    if (timer !== undefined) {
+      window.clearTimeout(timer);
+      saveTimersRef.current.delete(path);
+    }
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/web/src/CodeSurface.tsx:927]: Good catch — fixed in 17c3415: autosave PUTs are now tracked in an in-flight-promise map, and delete awaits the one for its path (if any) before issuing the DELETE call, so it can no longer race an already-fired autosave.

---
_Generated by [Claude Code](https://claude.ai/code)_
